examples/benchmark_preprocessing.py

```diff
@@ -0,0 +1,150 @@
+from braindecode.preprocessing import (
+    preprocess, Preprocessor, create_fixed_length_windows)
+
+
+###############################################################################
+# We create a function that goes through the usual three steps of data
+# preparation: (1) data loading, (2) continuous data preprocessing,
+# (3) windowing and (4) windowed data preprocessing. We use the
+# :class:`braindecode.datasets.SleepPhysionet` dataset for testing purposes.
+
+
+def prepare_data(n_recs, save, preload, n_jobs):
+    save_dir = tempfile.TemporaryDirectory().name if save else None
+
+    # (1) Load the data
+    concat_ds = SleepPhysionet(
+        subject_ids=range(n_recs), recording_ids=[1], crop_wake_mins=30,
+        preload=preload)
+    sfreq = concat_ds.datasets[0].raw.info['sfreq']
+
+    # (2) Preprocess the continuous data
+    preprocessors = [
+        Preprocessor(lambda x: x * 1e6),
+        Preprocessor('filter', l_freq=None, h_freq=30)
+    ]
+    preprocess(concat_ds, preprocessors, save_dir=save_dir, overwrite=True,
+               n_jobs=n_jobs)
+
+    # (3) Window the data
+    windows_ds = create_fixed_length_windows(
+        concat_ds, 0, 0, int(30 * sfreq), int(30 * sfreq), True,
+        preload=preload, n_jobs=n_jobs)
+
+    # Preprocess the windowed data
+    preprocessors = [Preprocessor(scale, channel_wise=True)]
+    preprocess(windows_ds, preprocessors, save_dir=save_dir, overwrite=True,
+               n_jobs=n_jobs)
+
+
+###############################################################################
+# Next, we can run our function and measure its run time and peak memory usage
+# for each one of our 4 cases above. We call the function multiple times with
+# each configuration to get better estimates.
+
+n_repets = 5
+all_n_recs = [8]  # Load and preprocess 8 recordings
+all_n_jobs = [1, 8]
+
+results = list()
+for _, n_recs, save, n_jobs in product(
+        range(n_repets), all_n_recs, [True, False], all_n_jobs):
+
+    start = time.time()
+    mem = max(memory_usage(
+        proc=(prepare_data, [n_recs, save, False, n_jobs], {})))
+    time_taken = time.time() - start
+
+    results.append({
+        'n_recs': n_recs,
+        'max_mem': mem,
+        'save': save,
+        'n_jobs': n_jobs,
+        'time': time_taken
+    })
+
+
+###############################################################################
+# Finally, we can plot the results:
+
+df = pd.DataFrame(results)
+ax = sns.scatterplot(
+    data=df, x='time', y='max_mem', style='n_jobs', hue='save',
+    palette='colorblind')
+ax.set_xlabel('Time (s)')
+ax.set_ylabel('Memory usage (MiB)')
+ax.set_title('Loading and preprocessing 8 recordings from Sleep Physionet')
+plt.show()
+
+
+###############################################################################
+# We see that parallel preprocessing without serialization (blue crosses) is
+# faster than simple sequential processing (blue circles), however it uses
+# significantly more memory.
```

Comment:
will this be clear on the doc built on CI?

Reply:
This raises a good point - this example is not currently run by the CI. I could adapt it so the computation time is shorter (it took around 20 mins on my machine if I remember well), but also I'm not sure whether the CI machines are configured to run with multiple cores?

Reply:
@robintibor I'm curious to know what you think - should we run this kind of example in the CI?

Reply:
I have mixed feeling here then. I think it's very valuable information but I am wondering how it will be perceived and discovered by users. We could keep it but if not run on CIs is likely to break silently. I would suggest to amend / update existing examples with note and warning sections to explain how the present example can make use of parallel and serialization should the data not fit in RAM.

Reply:
I am also not sure, in general alex suggestion makes sense to me but I am not sure if it makes sense to to this in all the existing examples, maybe just in the first one. For me, you could tryy to see if it can be made very fast and if it runs and produces examples on CI as expected? And if not, would be ok for me that it does not run on CI as well.

Reply:
Good news, I modified the example so that it runs on 2 files in parallel, and it yields the same conclusions as the previous version of the example on 8 files. Given that Github Actions runs on [2-core Azure machines](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources), we should be able to get similar results and include the example in the docs!